src/checks/series/spike_check.rs

```diff
@@ -0,0 +1,137 @@
+use crate::{DataCache, Error, Flag};
+
+/// Number of leading values a [`DataCache`] must contain to QC all its
+/// intended values with spike check
+pub const SPIKE_LEADING_PER_RUN: u8 = 1;
+/// Number of trailing values a [`DataCache`] must contain to QC all its
+/// intended values with spike check
+pub const SPIKE_TRAILING_PER_RUN: u8 = 1;
+
+/// Timeseries QC test that compares each observation against its immediate predecessor and
+/// successor.
+///
+/// The sum and difference of the differences between the observation and each of its neighbours
+/// is computed. the observation will be flagged as follows
+/// - If values are missing for the observation or either neighbour: DataMissing.
+/// - If the difference is less than 35% of the sum AND the sum is greater than `max`: Fail.
+/// - If the difference is less than 35% of the sum AND the sum is greater than `high`: Warn.
+/// - Else: Pass
+///
+/// Takes 3 datapoints, the second is the observation to be QCed, the first and third are needed
+/// to QC it.
+pub fn spike_check(data: &[Option<f32>; 3], max: f32) -> Flag {
+    if data.contains(&None) {
+        return Flag::DataMissing;
+    }
+    let data: Vec<f32> = data.iter().map(|opt| opt.unwrap()).collect();
```

Comment:
Would something like this be better or worse, opinions? I like it better because we don't have to deal with the indices later on
```suggestion
    let data: Vec<f32> = data.iter().map(|opt| opt.unwrap()).collect();
        let (right, obs, left) = match data[..] {
        [r, o, l] => (r.unwrap(), o.unwrap(), l.unwrap()),
        _ => unreachable!(),
    };
```

Reply:
done